src/main/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsController.java

```diff
@@ -57,4 +69,59 @@ public UserCommons getUserCommonsById(
     return userCommons;
   }
 
-}
\ No newline at end of file
+    @ApiOperation(value = "Buy a cow, totalWealth updated")
+  @PreAuthorize("hasRole('ROLE_USER')")
+  @PutMapping("/buy")
+  public ResponseEntity<String> putUserCommonsByIdBuy(
+          @ApiParam("commonsId") @RequestParam Long commonsId, 
+            @RequestBody @Valid UserCommons incomingUserCommons) throws JsonProcessingException {
+        User u = getCurrentUser().getUser();
+        Long userId = u.getId();
+
+        Commons commons = commonsRepository.findById(commonsId).orElseThrow( 
+          ()->new EntityNotFoundException(Commons.class, commonsId));
+        UserCommons userCommons = userCommonsRepository.findByCommonsIdAndUserId(commonsId, userId)
+        .orElseThrow(
+            () -> new EntityNotFoundException(UserCommons.class, "commonsId", commonsId, "userId", userId));
+
+        
+        long previousId = userCommons.getId();
+        incomingUserCommons.setId(previousId);
+        if(incomingUserCommons.getTotalWealth() >= commons.getCowPrice() ){
+          incomingUserCommons.setTotalWealth(incomingUserCommons.getTotalWealth() - commons.getCowPrice());
+          incomingUserCommons.setNumOfCows(incomingUserCommons.getNumOfCows() + 1);
+        }
+        userCommonsRepository.save(incomingUserCommons);
```

Comment:
instead of working with "incomingUserCommons", you'd work with just the `userCommons` that you looked up in the database.